components/camera/replaypcd/replaypcd_test.go

```diff
@@ -2,43 +2,397 @@ package replaypcd
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, pc, test.ShouldResemble, getPointCloudFromArtifact(t, i))
+		}
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("Calling NextPointCloud with end filter", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				End: "2000-01-01T12:00:10Z",
+			},
+		}
+
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		endTime, err := time.Parse(timeFormat, replayCamCfg.Interval.End)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Iterate through files that meet the provided filter
+		for i := 0; i < endTime.Second(); i++ {
+			pc, err := replayCamera.NextPointCloud(ctx)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, pc, test.ShouldResemble, getPointCloudFromArtifact(t, i))
+		}
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("Calling NextPointCloud with start filter", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				Start: "2000-01-01T12:00:05Z",
+			},
+		}
+
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		startTime, err := time.Parse(timeFormat, replayCamCfg.Interval.Start)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Iterate through files that meet the provided filter
+		for i := startTime.Second(); i < numPCDFiles; i++ {
+			pc, err := replayCamera.NextPointCloud(ctx)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, pc, test.ShouldResemble, getPointCloudFromArtifact(t, i))
+		}
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("Calling NextPointCloud with filter no data", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{
+			Source: "test",
```

Comment:
I think we should add tests showing that filtering on different values of source and robotID works too. Just to prove that our implementation of NextPointCloud is actually passing these values onto BinaryDataByFilter. Our implementation of BinaryDataByFilter could only accept a specific name/ID for source/robotID, for example, and return no results otherwise.